src/data/polynomial/coeff.lean

```diff
@@ -134,6 +134,11 @@ begin
   { exact λ h1, (h1 (nat.mem_antidiagonal.2 rfl)).elim }
 end
 
+theorem coeff_X_pow_mul (p : polynomial R) (n d : ℕ) :
+  coeff (polynomial.X ^ n * p) (d + n) = coeff p d :=
+by rw [(commute_X_pow p n).eq, coeff_mul_X_pow]
+
+@[simp]
```

Comment:
I'm not really convinced that a lemma that introduces an `ite` and nat subtraction is a good simp lemma

Reply:
Agreed